pkg/executor/adapter.go

```diff
@@ -2023,6 +2025,43 @@ func (a *ExecStmt) GetTextToLog(keepHint bool) string {
 	return sql
 }
 
+// getLazyText is equivalent to `a.GetTextToLog(false)`. Note that the s.Params is a shallow copy of
+// `sessVars.PlanCacheParams`, so you can only use the lazy text within the current stmt context.
+func (a *ExecStmt) getLazyStmtText() (s variable.LazyStmtText) {
+	sessVars := a.Ctx.GetSessionVars()
+	rmode := sessVars.EnableRedactLog
+	if rmode == errors.RedactLogEnable {
+		sql, _ := sessVars.StmtCtx.SQLDigest()
```

Comment:
Does L2034 and L2037 introduce memory allocation and copy, and is there any optimization space for these two functions called?